backend/src/main/kotlin/app/ehrenamtskarte/backend/userdata/webservice/UserImportHandler.kt

```diff
@@ -0,0 +1,112 @@
+            }
+            val file = files[0]
+
+            BufferedReader(InputStreamReader(file.content())).use { reader ->
+                getCSVParser(reader).use { csvParser ->
+                    importData(csvParser)
+                }
+            }
+            ctx.status(200).result("Import successfully completed")
+        } catch (exception: UserImportException) {
+            ctx.status(400).result(exception.message ?: "Invalid input provided")
+        } catch (exception: Exception) {
+            logger.error("Failed to perform user import", exception)
+            ctx.status(500).result("Internal error occurred")
+        }
+    }
+
+    private fun getCSVParser(reader: BufferedReader): CSVParser {
+        return CSVParser(
+            reader,
+            CSVFormat.DEFAULT.builder()
+                .setHeader()
+                .setTrim(true)
+                .setSkipHeaderRecord(true)
+                .build()
+        )
+    }
+
+    private fun importData(csvParser: CSVParser) {
+        val headers = csvParser.headerMap.keys
+        val requiredColumns = setOf("userHash", "startDate", "endDate", "revoked")
+        if (!headers.containsAll(requiredColumns)) {
+            throw UserImportException("Missing required columns: ${requiredColumns - headers}")
+        }
+
+        transaction {
+            // TODO as part of #1417: use projectId defined from the auth token
+            val project = ProjectEntity.find { Projects.project eq "showcase.entitlementcard.app" }.single()
+
+            for (entry in csvParser) {
+                if (entry.toMap().size != csvParser.headerMap.size) {
+                    throw UserImportException("The entry [${entry.joinToString(",")}] has missing data")
+                }
+
+                val userHash = entry.get("userHash").toByteArray()
+                if (!isValidUserHash(userHash)) {
+                    throw UserImportException("Invalid userHash format: ${userHash.joinToString(",")}")
+                }
+
+                val startDate = parseDate(entry.get("startDate"))
+                val endDate = parseDate(entry.get("endDate"))
+                if (startDate.isAfter(endDate)) {
+                    throw UserImportException("Start date [$startDate] cannot be after end date [$endDate]")
+                }
+
+                val revoked = entry.get("revoked").toBoolean()
+
+                UserEntitlementsRepository.insertOrUpdateUserData(
+                    userHash,
+                    startDate,
+                    endDate,
+                    revoked,
+                    project.id.value
+                )
+            }
+        }
+    }
+
+    private fun isValidUserHash(userHash: ByteArray): Boolean {
+        // TODO implement userHash validation after #1433
+        return true
+    }
+
+    private fun parseDate(dateString: String): LocalDate {
+        try {
+            return LocalDate.parse(dateString, DateTimeFormatter.ofPattern("dd.MM.yyyy"))
```

Comment:
If the format is not yet set in stone, I'd prefer if we'd use an [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date (as in "2024-12-24") (aka the most standard date format in all software)

Reply:
I would wait for @f1sh1918's reply as he gave me a sample data

Reply:
The sample data was not my decision but provided by the customer and can be found in nextcloud.
You can just ping @dkehne that he should inform the customer about that we want to use this date format here

Reply:
Leaving as is